### PR TITLE
more cleanup before generating CFG (continued)

### PR DESCRIPTION
**Pull request by yilongli**
do not review yet

### CODE REVIEW

**Comment by yilongli** [comment]
Jenkins: retest this please

**Comment by yilongli** [comment]
@andreistefanescu please review

File: java-backend/src/main/java/org/kframework/backend/java/symbolic/SymbolicRewriter.java [hunk 1]
```diff
@@ -151,18 +152,20 @@ public KRunState rewrite(ConstrainedTerm constrainedTerm, int bound) {
...
             }
 
-            return subject.unify(buildPattern(rule, subject.termContext()), rule.matchingInstructions(), rule.lhsOfReadCell(), rule.matchingVariables()).stream()
-                    .peek(s -> {
-                        RuleAuditing.succeed(rule);
-                        Coverage.print(subject.termContext().global().krunOptions.experimental.coverage, subject);
-                        Coverage.print(subject.termContext().global().krunOptions.experimental.coverage, rule);
-                    })
-                    .map(s -> Pair.of(buildResult(rule, s.getLeft(), subject.term(), !s.getRight()), rule))
+            return results = subject.unify(buildPattern(rule, subject.termContext()),
```
Comment (andreistefanescu) [inline java-backend/src/main/java/org/kframework/backend/java/symbolic/SymbolicRewriter.java:155]: break this in 2 lines

File: java-backend/src/main/java/org/kframework/backend/java/symbolic/SymbolicRewriter.java [hunk 2]
```diff
@@ -79,71 +76,75 @@ public KRunState rewrite(ConstrainedTerm constrainedTerm, int bound) {
...
-                        internalResults.add(any.get().get(0));
+                Set<Rule> rules = new LinkedHashSet<>(strategy.next());
+                rules.removeAll(failedRules);
+
+                Map<Rule, List<ConstrainedTerm>> rule2Results;
+                if (computeOne || !transition) {
+                    rule2Results = Collections.emptyMap();
+                    for (Rule rule : rules) {
+                        List<ConstrainedTerm> terms;
+                        if (!(terms = computeRewriteStepByRule(subject, rule)).isEmpty()) {
```
Comment (andreistefanescu) [inline java-backend/src/main/java/org/kframework/backend/java/symbolic/SymbolicRewriter.java]: split it on 2 lines; `=` and `==` are very easy to confuse, especially for someone with little java experience.

**Comment by andreistefanescu** [comment]
lgtm

**Comment by dwightguth** [comment]
This change seems to have increased the time to execute the Java Semantics test suite by a factor of 6. I suspect it's because of performance issues associated with Java 8 parallel streams use of ForkJoinPool.commonPool() but I don't really know for sure. In any case, @yilongli needs to investigate this and fix this issue otherswise we will have to revert this diff.

**Pull request merged**